### PR TITLE
In patch_sle we should wait a while to get the password.

### PR DESCRIPTION
**Pull request by hjluo**
in patch_sle we need to wait for a while to get the password, the following check for is-active network expect 'active' but it got the root password from the preceding step. the issue is change user and gpasswd cost too much time.

- Related ticket: https://progress.opensuse.org/issues/45515

- Verification run: http://openqa-apac1.suse.de/tests/3463

### CODE REVIEW

File: lib/opensusebasetest.pm [hunk 1]
```diff
@@ -483,6 +483,7 @@ sub wait_boot {
             type_line_svirt '', expect => $login_ready, timeout => $ready_time + 100, fail_message => 'Could not find login prompt';
             type_line_svirt "root", expect => 'Password';
             type_line_svirt "$testapi::password";
+            sleep 10;
```
Comment (lemon-suse) [inline lib/opensusebasetest.pm]: Please add comments here to describe why need add 10s sleep.

Comment (hjluo) [reply lib/opensusebasetest.pm]: added, thanks.

**Comment by okurz** [comment]
Hi, I think you were on a good track with "waiting a bit" before typing into the password prompt but sorry I reverted your PR but I did so for the following reasons:

* Just putting in a sleep with an arbitrary amount of time is *evil*. I have already seen the svirt s390x login to fail even with a sleep time of 10 seconds. In general sleep in coding is a bad sign, also see for example https://stackoverflow.com/questions/1096794/is-sleep-evil . When we are waiting for a specific event to happen then we should not just silently sleep but wait exactly for that: The specific event
* The "sleep 10" is a considerable waste of time for all the svirt based test runs and in general we should try to optimize our code base to invest as little time as necessary. Sorry but this change is not helping.
* Your PR contains two commits, both with the same subject line, both with the same misspelling. After all: You are changing a generic function that we use in multiple places. Just mentioning "patch_sle" is misleading
* The change about the serial permissions is unmotivated and not commented at all
* There was no review feedback from other teams yet and also did not ask for it even though you were changing generic methods which are used in nearly all tests.
* Your ticket as well as your PR does not hint to any statistics even though this is a sporadic issue. You simply can not sufficiently adress a sporadic issue with a single verification run, right?
* The ticket is far from the only one describing the issue and I highly suggest you look for better ticket relations to better understand the original issue and also so that we are able to provide a better fix which helps us all.

Every person with merge rights should keep the above in mind for any following pull requests and keep the quality of the overall code base in mind.

**Comment by hjluo** [comment]
Hi okurz,
Thanks for the comments.
This issue was not reproducible on our migration tests, we just want to put this on to test the sporadic issue. 
so I'd suggest if we can file a ticket for OpenQA tools while close this ticket. or can you provide us with a suggesting fix so we could avoid using the sleep in that step by means of "waiting for a specific event to happen"?

Huajian.Luo

**Comment by okurz** [comment]
Sure. I have linked the according – already existing – ticket in https://progress.opensuse.org/issues/45515 . It's https://progress.opensuse.org/issues/46394

File: lib/opensusebasetest.pm (hunk 1 above)
Comment (okurz) [reply lib/opensusebasetest.pm]: After looking at the above again I wonder how that should help. Why call `sleep 10` *after* typing the password?

**Comment by hjluo** [comment]
OK, thanks for the info.

**Pull request merged**